www/docs/content/recipes/digital-products.mdx

```diff
@@ -0,0 +1,1141 @@
+    customProps: {
+      icon: Icons['academic-cap-solid'],
+      description: 'Check out available official notification plugins.',
+    }
+  },
+  {
+    type: 'link',
+    href: 'https://medusajs.com/plugins/?filters=Storage&filters=Notification&categories=Storage',
+    label: 'Community File Service Plugins',
+    customProps: {
+      icon: Icons['academic-cap-solid'],
+      description: 'Check out available community notification service plugins.',
+    }
+  },
+  {
+    type: 'link',
+    href: '/development/notification/create-notification-provider',
+    label: 'Create a Notification Service',
+    customProps: {
+      icon: Icons['academic-cap-solid'],
+      description: 'Learn how to create a custom notification service.',
+    }
+  },
+]} />
+
+---
+
+## Create Custom Entity
+
+An entity represents a table in the database. The Medusa core defines entities necessary for the commerce features it provides. You can also extend those entities or create your own.
+
+To represent a digital product, it’s recommended to create an entity that has a relation to the `ProductVariant` entity. The `ProductVariant` entity represents the saleable variant of a `Product`.
+
+For example, if you’re selling the Harry Potter movies, you would have a `Product` with the name “Harry Potter”, and for each movie in the series a `ProductVariant`. Each `ProductVariant` would be associated with the custom entity you create that represents the downloadable movie.
+
+<DocCardList colSize={6} items={[
+  {
+    type: 'link',
+    href: '/development/entities/create',
+    label: 'Create an Entity',
+    customProps: {
+      icon: Icons['academic-cap-solid'],
+      description: 'Learn how to create an entity in the Medusa backend.',
+    }
+  },
+  {
+    type: 'link',
+    href: '/development/entities/extend-entity',
+    label: 'Extend an entity',
+    customProps: {
+      icon: Icons['academic-cap-solid'],
+      description: 'Learn how to extend an entity in the Medusa backend.',
+    }
+  },
+]} />
+
+<details>
+<summary>
+Example: Create ProductMedia Entity
+</summary>
+
+In this example, you’ll create a `ProductMedia` entity that is associated with the `ProductVariant` entity in a one-to-many relation.
+
+To do that, create the file `src/models/product-media.ts` with the following content:
+
+```ts title=src/models/product-media.ts
+import { 
+  BeforeInsert, 
+  Column, 
+  Entity,
+  JoinColumn,
+  ManyToOne,
+} from "typeorm"
+import { BaseEntity, ProductVariant } from "@medusajs/medusa"
+import { generateEntityId } from "@medusajs/medusa/dist/utils"
+
+export enum MediaType {
+  MAIN = "main",
+  PREVIEW = "preview"
+}
+
+@Entity()
+export class ProductMedia extends BaseEntity {
+  @Column({ type: "varchar" })
+  name: string
+
+  @Column ({ type: "enum", enum: MediaType, default: "main" })
+  type: MediaType
+
+  @Column({ type: "varchar" })
+  file_key: string
+
+  @Column({ type: "varchar" })
+  variant_id: string
+
+  @ManyToOne(
```

Comment:
Would avoid suggesting the foreign key constraint and instead just suggest a simple `variant_id: string` field for the relation.

This avoids creating dependencies at the DB level which will make composability difficult. (Note: this is more a future proofing thing given the current modularization work)